apps/src/blockly/addons/cdoFieldImageDropdown.js

```diff
@@ -10,10 +10,10 @@ export class CdoFieldImageDropdown extends FieldGridDropdown {
     // assuming that the number of options usually won't change that drastically,
     // so the number of columns can probably stay the same.
     const initialOptions = fixMenuGenerator(menuGenerator, width, height);
-    const numColumns =
-      initialOptions.length <= 7
-        ? 1
-        : Math.floor(Math.sqrt(initialOptions.length));
+    const numColumns = Math.max(
+      4,
+      Math.floor(Math.sqrt(initialOptions.length))
```

Comment:
Just curious - is it a standard practice to compute the # of cols using `Math.sqrt`?

Reply:
We are doing that to keep it approximately square shaped. Logic in the cdo-block repo that I am reproducing is [here](https://github.com/code-dot-org/blockly/pull/308/files)

Reply:
Thanks!